File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/SimpleContentTag.java
package com.mcreater.fxfluent.xaml.tags;

import com.mcreater.fxfluent.xaml.ResourceDict;
import org.dom4j.Element;

public abstract class SimpleContentTag<T> implements AbstractContentTag<T> {
    protected Element element;
    protected ResourceDict dict;
    public SimpleContentTag<T> writeElement(Element element) {
        this.element = element;
        return this;
    }

    public SimpleContentTag<T> setResourceDict(ResourceDict dict) {
        this.dict = dict;
        return this;
    }

    public Element getElement() {
        return element;
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/SimpleContentTag.kt
package com.mcreater.fxfluent.xaml.tags

import com.mcreater.fxfluent.xaml.ResourceDict
import org.dom4j.Element


abstract class SimpleContentTag<T> : AbstractContentTag<T> {
    protected var element: Element? = null
    var dict: ResourceDict? = null
    override fun writeElement(element: Element): SimpleContentTag<T> {
        this.element = element
        return this
    }

    override fun setResourceDict(dict: ResourceDict): SimpleContentTag<T> {
        this.dict = dict
        return this
    }

    override fun element(): Element {
        return element!!
    }

    override fun toString(): String {
        return toObject().toString()
    }
}
